### ISSUE TITLE
Posting dynamic type message to Web Worker.

### ISSUE THREAD

**Issue by ebala96** [dart-lang/web#242]
I have written dart side post message to send dart Map to web worker postMessage.

```
Future<void> postMessageDynamic(int id, message) async{
    final worker = _workers[id];
    Map<String , dynamic> messageMap = message as Map<String,dynamic>;

    worker.postMessage(messageMap.toJSBox);
}
```

But `toJsBox` creates an empty object without the actual Map data I want to send.
Let me know how can I send dart Map to postMessage without using json encoding.

**Comment by srujzs** [comment]
Dart `Map`s are not JS types and they can not be passed directly to `postMessage`. In order to clone a `Map` into a JS object, use [`jsify`](https://api.dart.dev/main/643b9ae8f74963a00255b9095373350b7a6b05a9/dart-js_interop/NullableObjectUtilExtension/jsify.html) e.g. `messageMap.jsify()`.

**Comment by ebala96** [comment]
A follow up on js interop.

How can I define list of any type inside interop definiftion?

**Comment by srujzs** [comment]
`List`s also are not JS types so they'll need to be converted to a `JSArray`, either through a manual conversion or using `jsify`. The appropriate type for a `JSArray` of any type is `JSArray<JSAny?>`.

**Comment by srujzs** [comment]
Right, the generic cast will fail because the type parameters are just `Object?`. JS arrays and objects are allowed to contain values of different types, and therefore the resulting generics are `Object?`. If you control the code that iterates through the map you can use the `cast` function to do the cast without having to iterate over the list e.g.

```dart
final map = (original.jsify().dartify() as Map).cast<String, List>();
for (final key in map.keys()) {
  final list = map[key].cast<String>();
  ...
}
```

If you want to avoid that extra iteration, you can manually convert the map yourself using interop (which will be faster), but that'll require more code.
